src/redux/reducers.js

```diff
@@ -673,6 +774,20 @@ export function posts(state = {}, action) {
     case response(ActionTypes.GET_SINGLE_POST): {
       return updatePostData(state, action)
     }
+    case ActionTypes.REALTIME_POST_NEW:
+    case ActionTypes.REALTIME_POST_UPDATE: {
+      return { ...state, [action.post.id]: postParser(action.post) }
+    }
+    case ActionTypes.REALTIME_COMMENT_NEW: {
+      const post = state[action.comment.postId]
+      if (!post || post.comments.indexOf(action.comment.id)!==-1) {
```

Comment:
If that is the first comment on the post, `post.comments` is undefined and there's JS error `Uncaught TypeError: Cannot read property 'indexOf' of undefined`. So we should either check it first or make sure it's always defined as `[]` (when we ingesting the posts on page load).